legacy/src/main/java/org/opensearch/sql/legacy/executor/join/NestedLoopsElasticExecutor.java

```diff
@@ -111,16 +112,25 @@ protected List<SearchHit> innerRun() throws SqlParseException {
           if (!BackOffRetryStrategy.isHealthy()) {
             throw new IllegalStateException("Memory circuit is broken");
           }
-          firstTableResponse =
-              client
-                  .prepareSearchScroll(firstTableResponse.getScrollId())
-                  .setScroll(new TimeValue(600000))
-                  .get();
+          Integer hintLimit = nestedLoopsRequest.getFirstTable().getHintLimit();
```

Comment:
For such change in legacy code, could you add comment in code to explain why?

Reply:
added comment.